ledger-service/http-json/src/test/scala/com/digitalasset/http/WebsocketServiceIntegrationTest.scala

```diff
@@ -106,6 +106,44 @@ class WebsocketServiceIntegrationTest
     }
   }
 
+  List(
+    SimpleScenario(
+      "query",
+      Uri.Path("/v1/stream/query"),
+      Source.single(TextMessage.Strict("""{"templateIds": ["AA:BB"]}"""))),
+    SimpleScenario(
+      "fetch",
+      Uri.Path("/v1/stream/fetch"),
+      Source.single(TextMessage.Strict("""[{"templateId": "AA:BB", "key": ["k", "v"]}]""")))
+  ).foreach { scenario =>
```

Comment:
Use `TableDrivenPropertyChecks` and `forEvery` for this kind of testing setup. That will still discriminate successful and failing cases without having to interpolate test names.

Reply:
`TableDrivenPropertyChecks` and `GeneratorDrivenPropertyChecks` do not work with `AsyncFreeSpec`
```
 found   : Unit
 required: scala.concurrent.Future[org.scalatest.compatible.Assertion]
```

Reply:
Oh well.

Reply:
yeah, it is either `AsyncFreeSpec` or `org.scalatest.concurrent.Futures#whenReady` at the end of every test scenario.